test/Conversion/ExportVerilog/pretty.mlir

```diff
@@ -0,0 +1,73 @@
+// RUN: circt-opt -export-verilog --split-input-file %s | FileCheck %s --strict-whitespace --match-full-lines
+
+sv.interface @IValidReady_Struct  {
+  sv.interface.signal @data : !hw.struct<foo: !hw.array<72xi1>, bar: !hw.array<128xi1>, baz: !hw.array<224xi1>>
+}
+
+// CHECK-LABEL:module structs({{.*}}
+//      CHECK:  assign _GEN =
+// CHECK-NEXT:    '{
+// CHECK-NEXT:      foo: ({_GEN_1, _GEN_0}),
+// CHECK-NEXT:      bar: ({_GEN_0, _GEN_0}),
+// CHECK-NEXT:      baz:
+// CHECK-NEXT:        ({
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_1,
+// CHECK-NEXT:           _GEN_0,
+// CHECK-NEXT:           _GEN_0
```

Comment:
Thanks for the feedback!  Let's iterate towards best output we can!

So braced list emission does this, and on some giant lists it can be ...sub-optimal.  I'd like to improve it especially for lists of elements that cannot break themselves (like example here).  Presently the comma-separated entries are put in a "cbox" which means all breaks within are broken or none are (not including breaks within subgroups).  Token stream is something like:

```
STRING({), BEGIN(CBOX), BREAK(0, INDENT), STRING(_GEN_1), STRING(,), BREAK(1), STRING(_GEN_2), STRING{,}, BREAK(1), ..., BREAK(0, -INDENT), END, STRING(})
```

Which means if the list doesn't fit, the group breaks together.   This is done to avoid output like:

```
a, b, c,
b + c + d
+ e, a + b
+ c
```

Which IMO is undesirable and hard to read.

---

Implementation-wise I'm not sure how best to create better output with the tools we have, but I'll think some more.
I tried making it so if one expression breaks the break after its comma is forced, but at the time couldn't make that work.

A few less-than-ideal ideas:

1. we further lean on the fact that we buffer tokens entirely when emitting any expression (for paren/cast insertion later) to scan the emitted tokens for breaks and switch to an "ibox" in that case.  Oof.

2. If the common-case simple expressions can be checked with a quick walk (maybe a short whitelist of operation types?) we could emit an "ibox" instead of the cbox (and no breaks next to braces) which does the "pack until line is full, break, continue to pack that line" behavior I think you're saying you prefer (if I I understand correctly).

FWIW both of these are operating under "let's pack these only if they're simple" idea, which maybe isn't best, LMK what behavior might be better!

There's likely other approaches/ideas, but does the above make sense? LMK what you think-- either in terms of preferred output or how to do it!

Reply:
Actually, what output are you saying you prefer? Can you show a short example (whatever demonstrates it, not necessarily one from the lit test)?  Thanks!

Reply:
I was wondering we can do something like this (in the future):
```
a, b, c, 
b + c + d 
+ e, 
a + b + c
```

Instead of inserting new lines to every operand:
```
a, 
b, 
c, 
b + c + d 
+ e, 
a + b + c
```

Reply:
Great, that's my preferred output too.  I'll think about how to do this.

Reply:
Thanks, I think we can improve this kind of details later so I'm happy to keep as is.